R/arc-select.R

```diff
@@ -461,3 +479,26 @@ count_results <- function(req, query) {
   RcppSimdJson::fparse(resp)[["count"]]
 }
 
+#' Validate fields
+#'
+#' [validate_fields()] ensures that fields passed to [arc_select()] match
+#' permissible values.
+#'
+#' @keywords internal
+#' @noRd
+match_fields <- function(fields, values = NULL, multiple = TRUE, error_call = rlang::caller_env()) {
+  if (is.null(fields) || identical(fields, "*")) {
+    return(fields)
+  }
```

Comment:
```suggestion
#' Match fields
#'
#' [match_fields()] ensures that fields passed to [arc_select()] match
#' permissible values.
#'
#' @keywords internal
#' @noRd
match_fields <- function(fields, values = NULL, multiple = TRUE, error_call = rlang::caller_env()) {
  if (is.null(fields) || identical(fields, "*")) {
    return(fields)
  }
```

What is returned if the if statement is evaluated to `TRUE` should `fields` should be returned?